### PR TITLE
More antag datum cleanup / code QoL improvements

### PR DESCRIPTION
**Pull request by SteelSlayer**
## What Does This PR Do
 This one is small this time. Hopefully the last one of these I'll be making.

Noteworthy changes:
- Removes `/datum/antagonist/proc/on_removal()`. The main/initial reason for this PR. This was essentially a wrapper for `Destroy()` and only served to needlessly complicate the cleaning up of things when the datum should be removed.
- Removes `/datum/antagonist/proc/specialization()`. This wasn't used for anything, isn't even used by any datum on TG, and I can't think of I use for this.
- Adds `var/needs_target` to `/datum/objective` and removes the `NO_TARGET_OBJECTIVES` list. I think this is preferable to keeping a separate list on the antag datum file itself. Also prevents unnecessary calls to `find_target()` when an objective never needs a target, like hijack.
- Fixes an issue where mindslaves were using `/datum/objective/protect` instead of the 
`/datum/objective/protect/mindslave` subtype which had logic added for it in #16947. Behavior added in that PR works now.
- Various other antag datum code cleanup / QoL improvments.

## Changelog
:cl:
fix: Mindslaves are now given the proper type of objective, which de-mindslaves them if their master is removed from the game via cryogenic storage
fix: When a mindslave is de-mindslaved, they no longer appear on the check antagonists page
/:cl:

### CODE REVIEW

File: code/modules/antagonists/traitor/datum_mindslave.dm [hunk 1]
```diff
@@ -21,10 +21,15 @@
...
+	..()
 
-/datum/antagonist/mindslave/Destroy()
+/datum/antagonist/mindslave/Destroy(force, ...)
+	if(owner.som)
+		owner.som.serv -= owner
+		owner.som.leave_serv_hud(owner)
 	master = null
+	SSticker.mode.implanted[owner] = null
+	SSticker.mode.implanted -= owner
```
Comment (hal9000PR) [inline code/modules/antagonists/traitor/datum_mindslave.dm:32]: might want to add this to the changelog as a fix. doesn't affect players, but it does affect admins.

**Review by hal9000PR — COMMENTED** [review]
only thing I could find, I will test the branch later. looks good so far

File: code/modules/antagonists/traitor/datum_mindslave.dm (hunk 1 above)
Comment (SteelSlayer) [reply code/modules/antagonists/traitor/datum_mindslave.dm:32]: Added.

File: code/modules/mob/living/silicon/login.dm [hunk 2]
```diff
@@ -1,11 +1,11 @@
 /mob/living/silicon/Login()
 	SetSleeping(0)
 	if(mind && SSticker && SSticker.mode)
+		mind.remove_all_antag_datums()
```
Comment (hal9000PR) [inline code/modules/mob/living/silicon/login.dm:4]: won't this cause issues with malf AI?

Comment (SteelSlayer) [reply code/modules/mob/living/silicon/login.dm:4]: Yeah potential issue here, I'll be looking into it.

Comment (SteelSlayer) [reply code/modules/mob/living/silicon/login.dm]: This was an issue. Reverted the change.

**Review by dearmochi — APPROVED** [review]

**Pull request merged**